### PR TITLE
Bootstrap retry

### PR DESCRIPTION
**Pull request by madadam**

### CODE REVIEW

File: src/core.rs [hunk 1]
```diff
@@ -61,6 +64,9 @@ const MAX_JOINING_NODES: usize = 1;
 /// of joining nodes.
 const JOINING_NODE_TIMEOUT_SECS: i64 = 5;
 
+/// Time (in seconds) after which bootstrap is cancelled (and possibly retried).
+const BOOTSTRAP_TIMEOUT_SECS: i64 = 10;
```
Comment (madadam) [inline src/core.rs:68]: The timeout might be again too optimistic, but it's easy to change later when necessary.

File: src/core.rs [hunk 2]
```diff
@@ -952,6 +968,13 @@ impl Core {
     }
 
     fn client_identify(&mut self, peer_id: PeerId) -> Result<(), RoutingError> {
+        if self.proxy_candidates.iter().any(|&(id, _)| id == peer_id) {
+            return Ok(());
```
Comment (afck) [inline src/core.rs:990]: Shouldn't we warn here that we are attempting to send a `ClientIdentify` to the same proxy twice?

Comment (madadam) [inline src/core.rs:990]: Probably a good idea. Will add it.

File: src/timer.rs [hunk 3]
```diff
@@ -0,0 +1,58 @@
+// Copyright 2015 MaidSafe.net limited.
```
Comment (afck) [inline src/timer.rs]: It's 2016. :-)

**Pull request merged**